static/partials/language-specific-deploy/go.md

```diff
@@ -1,137 +1,92 @@
+* It listens to the wild network `0.0.0.0`, not only `localhost` or `127.0.0.1`
+* It listens on port `8080`
+* You follow our build/run instructions
+
+In most cases you won't need to change anything to your application, except host/port and some configuration variables.
+
+### Modern Go project structure
+There are multiple ways to build/run a Go application, and this has evolved over the years. In its modern form a Go project can be a:
+- `Package`: one or more `.go` files you can `build` or `run`. `main` package and `main()` function are the default entry point
+- `Module`: one or more packages you can `install`, defined in a `go.mod` file (`go.sum` for checksums)
+- `Workspace`: one or more modules seamlessly combined, defined in a `go.work` file
+
+A module can be installed locally or from a remote repository if you pass its URL to the `install` command. A `Makefile` is sometimes used to define how a Go project is built, run and/or cleaned. The lightest form of a Go project is a `main.go` file to build. Some years ago, the `src/` folder was often used for source code, but using the `cmd/` folder instead is now a common practice. 
+
+If you want to limit from where a package can be imported it [should placed](https://docs.google.com/document/d/1e8kOo3r51b2BWtTs_1uADIA5djfXhPT36s6eHVRIvaU/edit) in a folder named `ìnternal/`. Access to functions in `.go` files is defined depending [on their name](https://go.dev/tour/basics/3): if it starts with a capital letter it's a public functions, if not it's a private function.
+
+For a complete project, a common files/folders organization can be:
+```
+application-root/
+├── go.work
+├── Makefile
+├── cmd/
+│   ├── main.go
+│   ├── other-file.go
+│   ├── other-package.go
+|   └── ...
+└── module/
+        ├── go.mod
+        ├── go.sum
+        ├── main.go
+        ├── other-module-file.go
+        └──internal/
+            └── internal-package/
+                └── internal-package-file.go
+```
+
+### Go build and deploy on Clever Cloud
+In such a situation, our strategy is to let the user choose how to build/run its application and make the deployment easy anyway. At first, we used the `goget` method, which is now deprecated. Thus, you can now use `gobuild` (for packages), `gomod` (for modules) or a configuration file. The latter will allow you to define a `Makefile` for custom build and a `main` executable to start the application.
+
+{{< callout type="info" >}}
+  If the required Go version declared in the `go.mod` is superior to the version built in the instance, it will be automatically updated.
+{{< /callout >}}
+
+#### Through Makefile and JSON configuration
+Create a file name `go.json` in a `clevercloud/` folder at the root of your repository. It will allow you to ask for a custom build through a `Makefile` ([learn more about it](https://en.wikipedia.org/wiki/Make_(software)#Makefiles)). For example with a single package in `main.go`:
+
+`/clevercloud/go.json`
```

Comment:
You can put a filename inside a code block now: https://imfing.github.io/hextra/docs/guide/syntax-highlighting/#filename